pkg/build/builder/common.go

```diff
@@ -66,6 +66,25 @@ func buildInfo(build *api.Build) []KeyValue {
 	return kv
 }
 
+// updateBuildStatus updates the current status of the build only if there was
+// an error
+func updateBuildStatus(c client.BuildInterface, build *api.Build) {
+	// NOTE: if reason is empty there's nothing to update, do i need this check
+	// here?
+	if build.Status.Reason == "" {
+		return
+	}
+
+	switch build.Status.Reason {
+	case api.StatusReasonPostCommitHookFailed:
+		build.Status.Message = "failed because of post commit hook"
```

Comment:
@bparees i thought we are adding these to help end users to diagnostic build failures. I don't  think having the messages as constants will be better than having real errors included in messages (for users). Tests can do `string.Contains()` and check for prefix, like `strings.HasPrefix("post commit hook returned error")`